src/main/java/InfrastructureManager/MatchMaking/MatchMaker.java

```diff
@@ -0,0 +1,84 @@
+package InfrastructureManager.MatchMaking;
+
+import InfrastructureManager.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class MatchMaker implements MasterInput, MasterOutput {
+
+    private MatchMakingAlgorithm algorithm;
+    private final ObjectMapper mapper;
+    private String command = "";
+
+    public MatchMaker() {
+        this.algorithm = new RandomMatchMaking(); // For now
+        this.mapper = new ObjectMapper();
+    }
+
+    public void setAlgorithm(MatchMakingAlgorithm algorithm) {
+        this.algorithm = algorithm;
+    }
+
+    private void assign (EdgeClient client) {
+        EdgeNode node = this.algorithm.match(client);
+        if (node == null) {
+            System.out.println("No node to assign");
+        } else {
+            Master.getInstance().mapClientNode(client, node);
```

Comment:
this coupling prevents us from running multiple instances of match-making, it would be better to have each instance of match-making be self-contained. Basically remove master use in MatchMaker and instead use lists as instance variables.

Reply:
Couple of questions here: 
- Do you mean remove all Master usage? Including for example `addClient` or `addNode` methods? Or just this one specifically?
- How would the replacement with lists look like? Could you maybe give me an example?

Reply:
Yes including addClient and addNode.

Here you are keeping 3 lists: clients, nodes and client-node mappings. These lists should be instance variables for each instance of the MatchMaker.

The Master should have at least 2 lists: clients and nodes (whether it has a mapping is up to you). For now you can use the same definition for client and node in the MatchMaker and the Master, but at some point it will be different. What am trying to say is that the Master sees the client from a different point of view than the MatchMaker, and thus needs to be independent.

For now, anytime a client is added to the Master it should also be added to the MatchMaker, we will need to change that later for specific uses cases, I just don't want to have a structure that is not flexible.